tests/phpunit/Civi/API/V4/ContactTest.php

```diff
@@ -0,0 +1,46 @@
+<?php
+namespace Civi\API\V4;
+use Civi\Api4\CustomGroup;
+use Civi\Api4\CustomField;
+use Civi\Api4\Contact;
+
+// fixme - what am I doing wrong to need this line?
+require_once 'UnitTestCase.php';
+
+/**
+ * @group headless
+ */
+class EntityTest extends UnitTestCase {
+
+  public function testGetWithCustomData() {
+    $customGroup = CustomGroup::create()
+      ->setCheckPermissions(FALSE)
+      ->setValue('name', 'CustomContactFields')
+      ->setValue('title', 'CustomContactFields')
+      ->setValue('extends', 'Contact')
```

Comment:
Like its predecessor, I still see the role of api4 as to smooth out the quirks of the BAO. So I think the fix is to override the generic `create` and `update` methods for this entity and fix the inputs.

Note that when we override a generic action, we have to annotate the api entity class so IDEs can do their static analysis correctly. [Here's an example of doing so](https://github.com/civicrm/api4/blob/d04e59530a78a740f8e76373fba31418415f4c55/Civi/Api4/Participant.php#L32-L32).